machida/lib/wallaroo/__init__.py

```diff
@@ -299,13 +298,28 @@ def partition(self, data):
                 return res
 
     # Case 3: Encoder
-    elif base_cls is Encoder:
+    elif base_cls is OctetEncoder:
         class C(base_cls):
             def encode(self, data):
                 return func(data)
 
+    elif base_cls is ConnectorEncoder:
+        class C(base_cls):
+            def encode(self, data, partition=None, sequence=None):
+                encoded = func(data)
+                if partition:
+                    part = str(partition).encode('utf-8')
+                else:
+                    part = b''
+                if sequence:
+                    seq = int(sequence)
+                else:
+                    seq = -1
+                meta = struct.pack('<H', len(part) + struct.calcsize('<q')) + part + struct.pack('<q', seq)
```

Comment:
Doing multiple string concatenations is going to be a performance hit in Python, since it creates intermediate strings after each concatenation. It would be less pretty, but I think more efficient for hot-path code like an encoder to do something like:
`meta = struct.pack('<H{}sq'.format(len(part)), len(part)+8, part, seq)`

Reply:
Yes. This was not intended to be optimized for the first release. I'll swap in the struct call since it should be easy to validate.